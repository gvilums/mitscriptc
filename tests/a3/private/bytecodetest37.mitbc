function
{
	functions =
	[
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions =
			[
				function
				{
					functions = [],
					constants = [None],
					parameter_count = 0,
					local_vars = [],
					local_ref_vars = [],
					free_vars = [x],
					names = [print],
					instructions = 
					[
						load_global	0
						push_ref	0
						load_ref
						call	1
						pop
						load_const	0
						return
					]
				}
			],
			constants = [None, 10],
			parameter_count = 0,
			local_vars = [x, child],
			local_ref_vars = [x],
			free_vars = [],
			names = [],
			instructions = 
			[
				load_const	0
				store_local	0
				load_const	0
				store_local	1
				load_const	1
				store_local	0
				load_func	0
				push_ref	0
				alloc_closure	1
				store_local	1
				load_local	1
				return
			]
		},
		function
		{
			functions = [],
			constants = [None],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
				load_const	0
				return
			]
		}
	],
	constants = [0, "forcing collection", 100000, 1, "done"],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [parent, q, f, n, z, print, input, intcast],
	instructions = 
	[
		load_func	0
		alloc_closure	0
		store_global	5
		load_func	1
		alloc_closure	0
		store_global	6
		load_func	2
		alloc_closure	0
		store_global	7
		load_func	3
		alloc_closure	0
		store_global	0
		load_global	0
		call	0
		store_global	1
		load_global	1
		call	0
		pop
		load_func	4
		alloc_closure	0
		store_global	2
		load_const	0
		store_global	3
		load_global	5
		load_const	1
		call	1
		pop
		load_global	3
		load_const	2
		swap
		gt
		not
		if	11
		load_global	2
		call	0
		pop
		alloc_record
		store_global	4
		load_global	3
		load_const	3
		add
		store_global	3
		goto	-15
		load_global	5
		load_const	4
		call	1
		pop
		load_global	1
		call	0
		pop
	]
}