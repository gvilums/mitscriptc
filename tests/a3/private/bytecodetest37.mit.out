10
forcing collection
done
10
